llvm/utils/TableGen/Common/PredicateExpander.cpp

```diff
@@ -152,19 +151,15 @@ void PredicateExpander::expandCheckOpcode(raw_ostream &OS,
     return;
   }
 
-  OS << '(';
-  ++Indent;
-  for (const Record *Rec : Opcodes) {
-    OS << '\n' << Indent;
-    if (!First)
-      OS << (shouldNegate() ? "&& " : "|| ");
-
-    expandCheckOpcode(OS, Rec);
-    First = false;
-  }
-
-  --Indent;
-  OS << '\n' << Indent << ')';
+  if (shouldNegate())
+    OS << "!";
+  OS << "llvm::is_contained(";
+  ListSeparator Sep;
+  OS << "{";
+  for (const Record *Inst : Opcodes)
+    OS << Sep << Inst->getValueAsString("Namespace") << "::" << Inst->getName();
+  OS << "}";
+  OS << ", MI" << (isByRef() ? "." : "->") << "getOpcode())";
```

Comment:
```suggestion
    OS << '!';
  OS << "llvm::is_contained(";
  ListSeparator Sep;
  OS << '{';
  for (const Record *Inst : Opcodes)
    OS << Sep << Inst->getValueAsString("Namespace") << "::" << Inst->getName();
  OS << '}';
  OS << ", MI" << (isByRef() ? "." : "->") << "getOpcode())";
```

Always use '' for single character prints, it calls a cheaper and inlined overload of <<